geomstats/geometry/discrete_curves.py

```diff
@@ -572,6 +621,35 @@ def dist(self, point_a, point_b, **kwargs):
 
         return dist
 
+    def space_derivative(self, curve):
+        """Compute space derivative of curve using centered differences.
+
+        Parameters
+        ----------
+        curve : array-like, shape=[..., n_points, ambient_dim]
+            Discrete curve.
+
+        Returns
+        -------
+        space_deriv : array-like, shape=[...,n_points, ambient_dim]
+        """
+        n_points = curve.shape[-2]
+        if n_points < 2:
+            raise ValueError('The curve needs to have at least 2 points.')
+
+        vec_1 = gs.array([-1.] + [0.] * (n_points - 2) + [1.])
+        vec_2 = gs.array([1. / 2] * (n_points - 2) + [1.])
+        vec_3 = gs.array([1.] + [1. / 2] * (n_points - 2))
+
+        mat_1 = from_vector_to_diagonal_matrix(vec_1, 0)
+        mat_2 = from_vector_to_diagonal_matrix(vec_2, -1)
+        mat_3 = from_vector_to_diagonal_matrix(vec_3, 1)
+        mat_space_deriv = mat_1 - mat_2 + mat_3
+
+        space_deriv = n_points * gs.matmul(mat_space_deriv, curve)
+
+        return space_deriv
+
 
 class ClosedDiscreteCurves(Manifold):
```

Comment:
This should probably be renamed `UnparameterizedClosedDiscreteCurves` then?
Alternatively, we could only have `UnparameterizedDiscreteCurves` and have `self.close_or_open` as an attribute? Maybe this is a discussion for after this PR, and we can just open an issue to think about the namings of these classes afterwards.